### PR TITLE
Added BosTau demographies from Boitard et al. (2016)

### PR DESCRIPTION
**Pull request by gregorgorjanc**
I have added four BosTau demographies from Boitard et al. (2016) https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005877 to complement the one from MacLeod et al. (2013) https://doi.org/10.1093/molbev/mst125

### CODE REVIEW

**Comment by gregorgorjanc** [comment]
Test fail because I have specified recombination rate specific to inferred demographic models - this functionality is available in PR https://github.com/popsim-consortium/stdpopsim/pull/1591

File: stdpopsim/catalog/BosTau/demographic_models.py [hunk 1]
```diff
@@ -105,3 +105,550 @@ def _HolsteinFriesian_1M13():
...
+
+def _HolsteinFriesian_1B16():
+    id = "HolsteinFriesian_1B16"
+    description = (
+        "Piecewise size model for Holstein-Friesian cattle (Boitard et al., 2016)."
+    )
+    long_description = """
+    The piecewise-constant population size model of Holstein-Friesian cattle
+    from Boitard et al. (2016). Effective population sizes were estimated using
+    Approximate Bayesian Computation approach with allele frequency spectrum and
```
Comment (petrelharp) [inline stdpopsim/catalog/BosTau/demographic_models.py:118]:
```suggestion
    Approximate Bayesian Computation with allele frequency spectrum and
```

File: stdpopsim/catalog/BosTau/demographic_models.py [hunk 2]
```diff
@@ -105,3 +105,550 @@ def _HolsteinFriesian_1M13():
...
+            reasons={stdpopsim.CiteReason.DEM_MODEL},
+        )
+    ]
+    return stdpopsim.DemographicModel(
+        id=id,
+        description=description,
+        long_description=long_description,
+        populations=populations,
+        citations=citations,
+        generation_time=_species.generation_time,
```
Comment (petrelharp) [inline stdpopsim/catalog/BosTau/demographic_models.py:149]: This is said above? Even if this agrees with the species definition, if it's specified in the model it should probably be put here? (However, does the generation time actually affect the model inference, as times are in generations? If not, maybe remove it from the parameter table above.)
```suggestion
        generation_time=5,
```

File: stdpopsim/catalog/BosTau/demographic_models.py [hunk 3]
```diff
@@ -105,3 +105,550 @@ def _HolsteinFriesian_1M13():
...
+_species.add_demographic_model(_HolsteinFriesian_1B16())
+
+
+def _Fleckvieh_1B16():
+    id = "Fleckvieh_1B16"
+    description = "Piecewise size model for Fleckvieh cattle (Boitard et al., 2016)."
+    long_description = """
+    The piecewise-constant population size model of Fleckvieh cattle
+    from Boitard et al. (2016). Effective population sizes were estimated using
+    Approximate Bayesian Computation approach with allele frequency spectrum and
```
Comment (petrelharp) [inline stdpopsim/catalog/BosTau/demographic_models.py:255]:
```suggestion
    Approximate Bayesian Computation with allele frequency spectrum and
```

File: stdpopsim/catalog/BosTau/demographic_models.py [hunk 4]
```diff
@@ -105,3 +105,550 @@ def _HolsteinFriesian_1M13():
...
+            reasons={stdpopsim.CiteReason.DEM_MODEL},
+        )
+    ]
+    return stdpopsim.DemographicModel(
+        id=id,
+        description=description,
+        long_description=long_description,
+        populations=populations,
+        citations=citations,
+        generation_time=_species.generation_time,
```
Comment (petrelharp) [inline stdpopsim/catalog/BosTau/demographic_models.py:286]: as above?

File: stdpopsim/catalog/BosTau/demographic_models.py [hunk 5]
```diff
@@ -105,3 +105,550 @@ def _HolsteinFriesian_1M13():
...
+_species.add_demographic_model(_Fleckvieh_1B16())
+
+
+def _Jersey_1B16():
+    id = "Jersey_1B16"
+    description = "Piecewise size model for Jersey cattle (Boitard et al., 2016)."
+    long_description = """
+    The piecewise-constant population size model of Jersey cattle
+    from Boitard et al. (2016). Effective population sizes were estimated using
+    Approximate Bayesian Computation approach with allele frequency spectrum and
```
Comment (petrelharp) [inline stdpopsim/catalog/BosTau/demographic_models.py:392]:
```suggestion
    Approximate Bayesian Computation with allele frequency spectrum and
```

File: stdpopsim/catalog/BosTau/demographic_models.py [hunk 6]
```diff
@@ -105,3 +105,550 @@ def _HolsteinFriesian_1M13():
...
+_species.add_demographic_model(_Jersey_1B16())
+
+
+def _Angus_1B16():
+    id = "Angus_1B16"
+    description = "Piecewise size model for Angus cattle (Boitard et al., 2016)."
+    long_description = """
+    The piecewise-constant population size model of Angus cattle
+    from Boitard et al. (2016). Effective population sizes were estimated using
+    Approximate Bayesian Computation approach with allele frequency spectrum and
```
Comment (petrelharp) [inline stdpopsim/catalog/BosTau/demographic_models.py:526]:
```suggestion
    Approximate Bayesian Computation with allele frequency spectrum and
```

**Review by petrelharp — CHANGES_REQUESTED** [review]
Looks good to me! Minor suggestions.

**Comment by petrelharp** [comment]
Don't worry about the pre-commit or failing macos tests; I think if you fix up those things in my review, and test it locally (as in [the docs](https://popsim-consortium.github.io/stdpopsim-docs/latest/development.html#testing-your-demographic-model-and-submitting-a-pr)) then we can merge this and you'll open QC issues.

Hm - QC will be a bit tricky here because of the "personal communication"; so, the personal communication should probably be pasted into the QC issue?

File: stdpopsim/catalog/BosTau/demographic_models.py (hunk 4 above)
Comment (gregorgorjanc) [reply stdpopsim/catalog/BosTau/demographic_models.py:286]: Will do.

File: stdpopsim/catalog/BosTau/demographic_models.py (hunk 2 above)
Comment (gregorgorjanc) [reply stdpopsim/catalog/BosTau/demographic_models.py:149]: Will do

**Comment by gregorgorjanc** [comment]
@petrelharp I have now went over all the suggested changes, done the testing in Python, so I think this is ready for a merge. 

Should I open 4 QC issues (I added 4 models from one publication) or 1 QC issue for the 4 models? I will add personal communication note in there ... I have made Simon Boitard (author of the publication) aware as well.

I created one issue here https://github.com/popsim-consortium/stdpopsim/issues/1598

**Review by petrelharp — APPROVED** [review]

**Comment by petrelharp** [comment]
Looks good - thanks! Now if we can get this QC'ed very soon (a week?) it'll go in the next release!

**Pull request merged**